r2/store.go

```diff
@@ -20,5 +20,36 @@
 
 package r2
 
-// Store is a construct that performs operations against a backing rule store.
-type Store interface{}
+import "github.com/m3db/m3metrics/rules"
+
+// Store is a construct that can perform operations against a backing rule store.
+type Store interface {
+	// Fetch namespaces and their version
+	FetchNamespaces() ([]string, int, error)
```

Comment:
minor nit: your utility functions in service go by `readNamespaces` and so on, here you're using `FetchNamespaces`. Pick one to stay consistent